Havit.Blazor.Components.Web.Bootstrap/Forms/HxMultiSelect.cs

```diff
@@ -202,7 +221,14 @@ private string GetInputText()
 		{
 			return InputText;
 		}
-		else if ((InputTextSelector is null) || (Data is null) || (CurrentValue is null))
+
+		// If filtering is enabled and the dropdown is visible then we want to display the user-entered filter text instead of the selected values summary
+		if (EnableFiltering && IsShown)
+		{
+			return InputText;
+		}
```

Comment:
(Oops, we misunderstood each other regarding the implementation. I will explain in general comment.)

Reply:
Thank you for the comments and feedback. I would still like to implement these changes and while I believe the way I've implemented the omni-box to be suitable for my scenario I understand it may not be suitable for everyone.
I shall rework this pull request within the next day or so to meet your standards and suggestions.

Reply:
I have reworked this component and believe it's in a good place to re-review.

Reply:
I've extended this to also include the select all functionality as requested in [Issue 453](https://github.com/havit/Havit.Blazor/issues/453).